### ISSUE TITLE
thumbsdb_256 and thumbsdb_128 get deleted over and over again

### ISSUE THREAD

**Issue by moxin3** [Mfarooq360/WinThumbsPreloader-V2#11]
I use GUI with admin and preload folder Icons, the issue is when I try to preload icons for more files folders etc my thumbsdb_256 and thumbsdb_128 get deleted or reset on auto

Is it possible someone make a tutorial for how to use GUI for maximum efficiency and faster scanning .

PS I use it on Windows 11

**Comment by VL4DST3R** [comment]
Only those two dbs get erased? The other sizes persist?

This may be related to an issue currently being investigated on #4 . Give one of the older versions of the tool a try and see if they behave the same.

**Comment by moxin3** [comment]
> Only those two dbs get erased? The other sizes persist?
> 
> This may be related to an issue currently being investigated on #4 . Give one of the older versions of the tool a try and see if they behave the same.

Yes it's exactly the same issue, so what you suggest last stable release I shall use or any beta version you recommend, and which is more efficient way to use 

From form button or right click menu on folder

Additionally I have 8 threads on CPU or logical processors I use 6 for the preloader program but it won't use much resources

**Comment by VL4DST3R** [comment]
> what you suggest last stable release I shall use or any beta version you recommend

There are no "stable" releases for V2 yet, try beta1 for starters, since its the furthest away and see what differences are there for your use case, then we go on from there. As to how to use it- I don't think it matters, use whatever is more convenient for you, ui/cli, the point of it is to test if those dbs persist or not.

> Additionally I have 8 threads on CPU or logical processors I use 6 for the preloader program but it won't use much resources

This may also be a case of issue #10

**Comment by Mfarooq360** [comment]
> > Additionally I have 8 threads on CPU or logical processors I use 6 for the preloader program but it won't use much resources
> 
> This may also be a case of issue #10
>
Is the multithreading problem occurring for you as well? 

So far what I know about the way it works is that even if the disk isn't the bottleneck, the program may decide not to use the full CPU for other reasons (potentially another bottleneck or maxing out the speed of windows thumbnail generation perhaps), though from what I'm hearing from that other issue, it's barely using any CPU on an SSD (5% CPU usage) which is interesting behavior that I haven't discovered yet, unless the processor being used has significantly more cores.

In my testing, it's normal for my CPU to use about 50-80% with an 8 core processor on the default threading settings and the disk usage and thumbnail processing speed is significantly faster than the previous forks which spawned separate instances of the program for every file, which just used a lot of the CPU up without any benefit.

**Comment by VL4DST3R** [comment]
> Is the multithreading problem occurring for you as well?

I would say no, for me the results are satisfactory on ssds as well, with noticeable usage on all cores *at least when when pre-existng thumbs are being checked* (i have them all already generated at the moment).

* Here is a graph of the results of 2 jobs ran on an ssd: 
![image](https://github.com/Mfarooq360/WinThumbsPreloader-V2/assets/32654378/a5e6b210-d8c2-4bf4-bce0-a91d2fe52146)
LEFT GRAPH: Purple = target SSD Red = System SSD (.db checking)
(light-purple indicates usage, solid graph indicates transfer speed)
RIGHT GRAPH: CPU threads (12c/24t)


For hard drives this is not an issue either since pretty much always the bottleneck is their "active time" (read speed):

* Here is an example with another 2 jobs, this time on a HDD (different files):
![image](https://github.com/Mfarooq360/WinThumbsPreloader-V2/assets/32654378/85c6a106-48af-44e8-aa88-719ab454c74d)
LEFT GRAPH: Yellow = target HDD Red = System SSD (.db checking)
(light-yellow indicates usage, solid graph indicates transfer speed)
RIGHT GRAPH: CPU threads (12c/24t)

## However,

Remember that my "regular" usage is a bit different from what I imagine most people do.

I have *a* bat file that orchestrates multiple (4 at the time of writing) individual bat files which get run as separate processes, one for each physical drive on my pc.

![image](https://github.com/Mfarooq360/WinThumbsPreloader-V2/assets/32654378/c44e5c40-5e4d-4fdb-a997-8ba349db6166)

Each of them goes through a list of paths and executes whatever folder thumb preloading is needed for that disk *sequentially* (with the -m flag for multithreading), since I figured there is no point in more parallelization if the disk itself is the bottleneck.

```
{snip}

SET WINTHUMBS=E:\Programs\WinThumbsPreloader\WinThumbsPreloader2b4.exe

SET PATH1=-r -m "S:\Automatic1111\webui\outputs\"
SET PATH2=-m "S:\ComfyUI\ComfyUI\output"
SET PATH3=-r -m "S:\Training Sets"

:loop

{snip}
```

**Comment by moxin3** [comment]
> Only those two dbs get erased? The other sizes persist?
> 
> This may be related to an issue currently being investigated on #4 . Give one of the older versions of the tool a try and see if they behave the same.

thumbcache_256 & thumbcache_1280 gets deleted rest are too small to noticei switched to v1.1.0 its noticeably faster but deletion of cache occurs i think some kind of files perhaps text files or program files causes this to crash and reset but iam not sure. Anyway this piece of software is too fun to use

**Comment by moxin3** [comment]
https://github.com/Mfarooq360/WinThumbsPreloader-V2/issues/10#issuecomment-1913138368
